src/java.base/share/classes/java/lang/Process.java

```diff
@@ -456,14 +457,54 @@ public final BufferedWriter outputWriter(Charset charset) {
      * @since 1.8
      */
     public boolean waitFor(long timeout, TimeUnit unit)
-        throws InterruptedException
-    {
-        long remainingNanos = unit.toNanos(timeout); // throw NPE before other conditions
+        throws InterruptedException {
+        Objects.requireNonNull(unit, "unit"); // throw NPE before other conditions
+
         if (hasExited())
             return true;
         if (timeout <= 0)
             return false;
 
+        return waitForNanos(unit.toNanos(timeout));
+    }
+
+    /**
+     * Causes the current thread to wait, if necessary, until the
+     * process represented by this {@code Process} object has
+     * terminated, or the specified waiting duration elapses.
+     *
+     * <p>If the process has already terminated then this method returns
+     * immediately with the value {@code true}.  If the process has not
+     * terminated and the duration is not positive, then
+     * this method returns immediately with the value {@code false}.
+     *
+     * <p>The default implementation of this method polls the {@code exitValue}
```

Comment:
There was discussion in the CSR on moving this to an `@implSpec`, are you going to do that change?

Reply:
It will be addressed by https://bugs.openjdk.org/browse/JDK-8336679

Reply:
I am planning to add `@implSpec` with a separate issue: [JDK-8336679](https://bugs.openjdk.org/browse/JDK-8336679)

Reply:
Okay, just a bit strange to add waitFor(Duration) here and then follow-up immediately to change it to move the text to an implSpec, I had hoped it would be done here.

Reply:
I believe the choice was made because both waitFor methods need this documentation update; better not touch the (long,TimeUnit) one in this patch.